codec/encoder/core/src/encoder.cpp

```diff
@@ -544,7 +544,61 @@ void DumpRecFrame (SPicture* pCurPicture, const char* kpFileName, const int8_t k
   }
 }
 
+/*!
+ * \brief   Copy the reconstructed picture to SFrameBSInfo structure
+ */
+void CopyReconPicToBSInfo (sWelsEncCtx* pCtx, SFrameBSInfo* pFbi, SPicture* pSrcRecPic, const SSourcePicture* pSrcPic) {
+  if (NULL != pFbi->pReconPic->pData[0] || NULL != pFbi->pReconPic->pData[1] || NULL != pFbi->pReconPic->pData[2]) {
```

Comment:
This is more a personal liking by I prefer early returns since they improve readabilidy of code:

```
if (NULL = pFbi->pReconPic->pData[0] && NULL == pFbi->pReconPic->pData[1] || NULL == pFbi->pReconPic->pData[2]) {
  pFbi->bHaveRecon = false;
  return;
}

/* Do the actual work */

 ```